packages/hash/frontend/src/pages/login.page.tsx

```diff
@@ -1,39 +1,212 @@
 
-const Page: NextPageWithLayout = () => {
-  const { refetch } = useUser();
+const LoginPage: NextPageWithLayout = () => {
+  // Get ?flow=... from the URL
   const router = useRouter();
 
-  return (
-    <LoginModal
-      show
-      onLoggedIn={({ accountSignupComplete, accountId }) => {
-        void refetch().then(() => {
-          // redirect to invite page if login occured from invitation link
-          if (accountSignupComplete) {
-            if (
-              isParsedInvitationEmailQuery(router.query) ||
-              isParsedInvitationLinkQuery(router.query)
-            ) {
-              void router.push({ pathname: "/invite", query: router.query });
+  const {
+    return_to: returnTo,
+    flow: flowId,
+    // Refresh means we want to refresh the session. This is needed, for example, when we want to update the password
+    // of a user.
+    refresh,
+    // AAL = Authorization Assurance Level. This implies that we want to upgrade the AAL, meaning that we want
+    // to perform two-factor authentication/verification.
+    aal,
+  } = router.query;
+
+  const [flow, setFlow] = useState<SelfServiceLoginFlow>();
+
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const handleFlowError = useMemo(
+    () =>
+      createFlowErrorHandler({
+        router,
+        flowType: "login",
+        setFlow,
+        setErrorMessage,
+      }),
+    [router, setFlow, setErrorMessage],
+  );
+
+  // This might be confusing, but we want to show the user an option
+  // to sign out if they are performing two-factor authentication!
+  const { logout } = useLogoutFlow([aal, refresh]);
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    // If ?flow=.. was in the URL, we fetch it
+    if (flowId) {
+      oryKratosClient
+        .getSelfServiceLoginFlow(String(flowId))
+        .then(({ data }) => setFlow(data))
+        .catch(handleFlowError);
+      return;
+    }
+
+    // Otherwise we initialize it
+    oryKratosClient
+      .initializeSelfServiceLoginFlowForBrowsers(
+        Boolean(refresh),
+        aal ? String(aal) : undefined,
+        returnTo ? String(returnTo) : undefined,
+      )
+      .then(({ data }) => setFlow(data))
+      .catch(handleFlowError);
+  }, [
+    flowId,
+    router,
+    router.isReady,
+    aal,
+    refresh,
+    returnTo,
+    flow,
+    handleFlowError,
+  ]);
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (event) => {
+    event.preventDefault();
+
+    if (!flow || !email || !password) {
+      return;
+    }
+
+    const csrf_token = mustGetCsrfTokenFromFlow(flow);
+
+    void router
+      // On submission, add the flow ID to the URL but do not navigate. This prevents the user loosing
+      // his data when she/he reloads the page.
```

Comment:
```suggestion
      // their data when they reload the page.
```

If you're trying to go for gender neutral